packages/runtime/__tests__/ImplWrapper/ImplWrapper.spec.tsx

```diff
@@ -132,3 +133,46 @@ describe('when component merge state asynchronously', () => {
     clearTesterMap();
   });
 });
+
+describe('slot trait if condition', () => {
+  it('only teardown component state when it is not hidden before the check', () => {
+    const { App, stateManager, apiService } = initSunmaoUI({ libs: [TestLib] });
+    stateManager.noConsoleError = true;
+    const { unmount } = render(<App options={TabsWithSlotsSchema} />);
+    expect(screen.getByTestId('tabs')).toHaveTextContent(`Tab OneTab Two`);
+
+    act(() => {
+      apiService.send('uiMethod', {
+        componentId: 'input',
+        name: 'setValue',
+        parameters: 'new-value',
+      });
+    });
+    expect(stateManager.store).toMatchInlineSnapshot(`
+      Object {
+        "input": Object {
+          "value": "new-value",
+        },
+        "tabs": Object {
+          "selectedTabIndex": 0,
+        },
+      }
+    `);
+    act(() => {
+      screen.getByTestId('tabs-tab-1').click();
+    });
+    expect(stateManager.store).toMatchInlineSnapshot(`
+      Object {
+        "input": Object {
+          "value": "new-value",
```

Comment:
Without the fix, here will be `value: ""`, because the state has been teardown and re-init.